dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -318,16 +320,45 @@ def runCrossValidation(self, jobId, interfaces):
         failures.extend(Validator.crossValidateSql(rules, submissionId))
         bucketName = CONFIG_BROKER['aws_bucket']
         regionName = CONFIG_BROKER['aws_region']
-        errorFileName = self.getFileName(interfaces.jobDb.getCrossFileReportPath(submissionId))
         errorDb = interfaces.errorDb
 
-        with self.getWriter(regionName, bucketName, errorFileName,
-                            self.crossFileReportHeaders) as writer:
-            for failure in failures:
-                writer.write(failure)
-                errorDb.recordRowError(jobId, "cross_file",
-                    failure[0], failure[2], failure[4], failure[5])
-            writer.finishBatch()
+        # Get list of file types
+        fileTypes = interfaces.validationDb.getFileTypeList()
+        # Create dict mapping file types to empty lists
+        fileTypeDict = {targetFile:[] for targetFile in fileTypes}
+
+        # Sort failures by source and target file
+        failureDict = {sourceFile:copy.deepcopy(fileTypeDict) for sourceFile in fileTypes}
+
+        for failure in failures:
+            errorDb.recordRowError(jobId, "cross_file",
+                    failure[0], failure[2], failure[4], failure[5], failure[6], failure[7])
```

Comment:
Do you think it would be helpful to include the target file in the error reports as well? I know you can infer that from the file name, and it would be repetitive, but I'm wondering if we'd ever want a sure way to get the target file w/o having to parse the name of the error report. Also, though the front-end may not need that name, as someone who opened an error report to review it, I think it would be helpful to have the target file listed explicitly.